src/journal/FutureImpl.cc

```diff
@@ -61,13 +61,13 @@ void FutureImpl::flush(Context *on_safe) {
 }
 
 FutureImplPtr FutureImpl::prepare_flush(FlushHandlers *flush_handlers) {
-  Mutex::Locker locker(m_lock);
+  std::lock_guard locker{m_lock};
   return prepare_flush(flush_handlers, m_lock);
 }
 
 FutureImplPtr FutureImpl::prepare_flush(FlushHandlers *flush_handlers,
-                                        Mutex &lock) {
-  ceph_assert(m_lock.is_locked());
+                                        std::mutex &lock) {
```

Comment:
```ceph::mutex```?

Reply:
it's intended. as `FutureImpl::m_lock` is now `std::mutex`.

Reply:
If we're wanting to check that the mutex is locked by *somebody* but not necessarily us, I'd argue that's an antipattern and we shouldn't add methods to support it.

Reply:
@adamemerson I disagree with your assertion. While it's not going to catch bugs during integration, it does catch errors during unit tests (which we have hundreds of) and it acts like a comment to describe the expected state invariants.

Reply:
in this specific case where prepare_flush() has an overload taking the mutex to document that it requires the lock, that mutex argument could instead be a std::lock_guard to represent a -held- lock. that pattern not only documents the expectation, but allows the compiler to verify it for us at compile time - without requiring the mutex itself to track whether it's locked. when the lock type is a std::unique_lock, you don't get the compile-time check, but you can still assert on lock.owns_lock()

Reply:
That's what I was getting at with the question of 'is_locked' versus 'is_locked_by_me'.

If what we're getting at is the latter, I'd just pass in a reference to a lock_guard or unique_lock, the way Casey suggests.

Reply:
@cbodley thank you. i think passing a `std::lock_guard` is way better. probably we can make it so in a follow-up PR to do the locks and checks in a more idiomatic way.